py_mcai_worker_sdk/src/main.rs

```diff
@@ -132,158 +114,192 @@ impl MessageEvent for PythonWorkerEvent {
-      let result = job_result
-        .with_status(JobStatus::Error)
-        .with_message(&error);
-      return Err(MessageError::ProcessingError(result));
+fn get_python_module<'a>(gil: &'a GILGuard) -> Result<(Python<'a>, &'a PyModule)> {
+  let python_file_content = PythonWorkerEvent::read_python_file();
+  let py = gil.python();
+  let python_module = PyModule::from_code(py, &python_file_content, "worker.py", "worker")
+    .map_err(|error| {
+      MessageError::RuntimeError(format!(
+        "unable to create the python module: {}",
+        py_err_to_string(py, error)
+      ))
+    })?;
+  Ok((py, python_module))
+}
+
+fn call_module_function<'a>(
+  py: Python,
+  python_module: &'a PyModule,
+  function_name: &'a str,
+  args: impl IntoPy<Py<PyTuple>>,
+) -> std::result::Result<&'a PyAny, String> {
+  match python_module.call1(function_name, args) {
+    Ok(response) => Ok(response),
+    Err(error) => {
+      let stacktrace = if let Some(tb) = &error.ptraceback {
+        let traceback = py.import("traceback").unwrap();
+        let locals = [("traceback", traceback)].into_py_dict(py);
+
+        locals.set_item("tb", tb).unwrap();
+
+        py.eval("traceback.format_tb(tb)", None, Some(locals))
+          .expect("Unknown python error, unable to get the stacktrace")
+          .to_string()
+      } else {
+        "Unknown python error, no stackstrace".to_string()
+      };
+
+      let error_msg = py_err_to_string(py, error);
+      let error_message = format!("{}\n\nStacktrace:\n{}", error_msg, stacktrace);
+      Err(error_message)
     }
+  }
+}
 
-    let callback_handle = CallbackHandle {
-      channel,
-      job: job.clone(),
-    };
+impl MessageEvent<PythonWorkerParameters> for PythonWorkerEvent {
+  fn get_name(&self) -> String {
+    self.get_string_from_module("get_name")
+  }
+
+  fn get_short_description(&self) -> String {
+    self.get_string_from_module("get_short_description")
+  }
 
-    match python_module.call1("process", (callback_handle, list_of_parameters)) {
-      Ok(response) => {
-        if let Some(mut destination_paths) = get_destination_paths(response) {
-          job_result = job_result.with_destination_paths(&mut destination_paths);
-        }
+  fn get_description(&self) -> String {
+    self.get_string_from_module("get_description")
+  }
 
-        Ok(job_result.with_status(JobStatus::Completed))
-      }
-      Err(error) => {
-        let stacktrace = if let Some(tb) = &error.ptraceback {
-          let locals = [("traceback", traceback)].into_py_dict(py);
+  fn get_version(&self) -> Version {
+    Version::parse(&self.get_string_from_module("get_version"))
+      .expect("unable to parse version (please use SemVer format)")
+  }
 
-          locals.set_item("tb", tb).unwrap();
+  fn init(&mut self) -> Result<()> {
+    let gil = Python::acquire_gil();
+    let (py, python_module) = get_python_module(&gil)?;
+
+    let _result = call_module_function(py, python_module, "init", ())
```

Comment:
check if the method is not present, just continue.
Here the init is required, not as in Rust worker.